From c9c3ebe0be70653da8fb8c21381aa35d6e2cede4 Mon Sep 17 00:00:00 2001
From: Tianmi Chen <tianmi.chen@intel.com>
Date: Thu, 2 May 2013 16:58:40 +0800
Subject: libstagefright: Ignore codec private data to make more clips playable

BZ: 98517

In some MKV files, there is no codec private data for MPEG4 written in
the container. Thus media mime type or video dimension will not be set
in current logic and media framework will treat these clips as malformed
files. Actually, clips can be played fine even without these private data.
To get it fixed, just simply ignore the check of the private data.

Category: aosp improvement
Domain: Video.Media-container
Origin: internal
Upstream-Candidate: yes

Change-Id: I0f163317ab3e8b66c3442b001f333d7399de439e
Orig-Change-Id: I3fc2634b93c8c4eabf5197e0f333ebe602862f7d
Signed-off-by: Tianmi Chen <tianmi.chen@intel.com>
Signed-off-by: Dan Liang <dan.liang@intel.com>
Signed-off-by: Yuanjun Huang <yuanjun.huang@intel.com>
---
 .../libstagefright/matroska/MatroskaExtractor.cpp  |    8 +++-----
 1 files changed, 3 insertions(+), 5 deletions(-)

diff --git a/media/libstagefright/matroska/MatroskaExtractor.cpp b/media/libstagefright/matroska/MatroskaExtractor.cpp
index 09c342e..18be213 100644
--- a/media/libstagefright/matroska/MatroskaExtractor.cpp
+++ b/media/libstagefright/matroska/MatroskaExtractor.cpp
@@ -863,15 +863,13 @@ void MatroskaExtractor::addTracks() {
                     meta->setCString(kKeyMIMEType, MEDIA_MIMETYPE_VIDEO_AVC);
                     meta->setData(kKeyAVCC, 0, codecPrivate, codecPrivateSize);
                 } else if (!strcmp("V_MPEG4/ISO/ASP", codecID)) {
+                    meta->setCString(kKeyMIMEType, MEDIA_MIMETYPE_VIDEO_MPEG4);
                     if (codecPrivateSize > 0) {
-                        meta->setCString(
-                                kKeyMIMEType, MEDIA_MIMETYPE_VIDEO_MPEG4);
                         addESDSFromCodecPrivate(
                                 meta, false, codecPrivate, codecPrivateSize);
                     } else {
-                        ALOGW("%s is detected, but does not have configuration.",
-                                codecID);
-                        continue;
+                        ALOGW("%s is detected, but does not have configuration.", codecID);
+                        //continue;
                     }
                 } else if (!strcmp("V_VP8", codecID)) {
                     meta->setCString(kKeyMIMEType, MEDIA_MIMETYPE_VIDEO_VP8);
-- 
1.7.4.1

